projects/igniteui-angular/src/lib/action-strip/action-strip.component.ts

```diff
@@ -139,6 +148,17 @@ export class IgxActionStripComponent extends DisplayDensityBase {
     @ViewChild('dropdown')
     private menu: IgxDropDownComponent;
 
+    /**
+     * @hidden
+     * @internal
+     */
+    ngAfterContentInit() {
+        this.menuItems = [... this._menuItems.toArray()];
+        this._menuItems.changes.subscribe(x => {
+            this.menuItems = [... this._menuItems.toArray()];
```

Comment:
I am not sure if this is a valid scenario but if the `_menuItems` collection changes, wouldn't that override any buttons added before that when the `asMenuItems` is true?